src/OrchardCore.Modules/OrchardCore.Media/Processing/MediaResizingFileProvider.cs

```diff
@@ -88,19 +83,22 @@ public bool IsValidRequest(HttpContext context)
         }
 
         /// <inheritdoc/>
-        public async Task<IImageResolver> GetAsync(HttpContext context)
+        public Task<IImageResolver> GetAsync(HttpContext context)
         {
-            // Path has already been correctly parsed before here.
-            var filePath = _mediaStore.MapPublicUrlToPath(context.Request.PathBase + context.Request.Path.Value);
+            // Remove assets request path.
+            var path = context.Request.Path.Value.Substring(_assetsRequestPath.Length);
+
+            var fileInfo = _mediaFileProvider.GetFileInfo(path);
```

Comment:
Everything now arrives at ImageSharp via an `IFileProvider` - either cache if using blob, or standard App_Data/Media folder if not. So nothing to change here for a new storage provider.